File: 2021/day2/part1.pl
#!/usr/bin/perl
use feature "say";
open (my $fh, "<", "input") or die "Can't open file";
my $vert = 0;
my $hori = 0;
while (<$fh>) {
    chomp $_;
    my @temp = split(" ", $_);
    if ($temp[0] eq "up") { 
        $vert -= $temp[1];
    }
    elsif ($temp[0] eq "down") {
        $vert += $temp[1];
    }
    else {
        $hori += $temp[1];
    }
}
say $hori * $vert;

File: 2021/day2/part1.py
#!/usr/bin/python3
vert = 0
hori = 0
with open("input", "r") as filehandle:
    for line in filehandle.readlines():
        line = line.strip()
        opt, dist = line.split(" ")
        if (opt == "up"):
            vert -= int(dist)
        elif (opt == "down"):
            vert += int(dist)
        else:
            hori += int(dist)
print(vert * hori)
